crates/katana/primitives/src/genesis/json.rs

```diff
@@ -312,34 +320,99 @@ impl TryFrom<GenesisJson> for Genesis {
                                 serde_json::from_value(artifact.clone())?;
                             casm.class_hash()?
                         };
-
                         (class_hash, class_hash, None, Arc::new(CompiledContractClass::V0(casm)))
                     }
                 };
 
+                // Add the class name and class hash to the mapping
+                let _ = name.map(|name| {
+                    name_to_class_hash.insert(name, class_hash);
+                });
+
                 Ok((class_hash, GenesisClass { compiled_class_hash, sierra, casm }))
             })
             .collect::<Result<_, GenesisJsonError>>()?;
 
+        // Populate the classes and name_to_class_hash
         let mut fee_token = FeeTokenConfig {
             name: value.fee_token.name,
             symbol: value.fee_token.symbol,
             total_supply: U256::zero(),
             decimals: value.fee_token.decimals,
             address: value.fee_token.address.unwrap_or(DEFAULT_FEE_TOKEN_ADDRESS),
-            class_hash: value.fee_token.class.unwrap_or(DEFAULT_LEGACY_ERC20_CONTRACT_CLASS_HASH),
+            class_hash: match value.fee_token.class {
+                Some(NameOrHash::ClassHash(class_hash)) => class_hash,
+                Some(NameOrHash::ClassName(class_name)) => *name_to_class_hash
+                    .get(&class_name)
+                    .ok_or_else(|| value_out_of_range_error::ValueOutOfRangeError)?,
```

Comment:
Here we want to return a new error something like:
```rust
#[error("Class name not found in the genesis classes: {0}")]
InvalidClassName(String)
```
You should add this to the `GenesisJsonError`.

This applies to all conversions from name that fails.